Prod 649: Give specific user credit admin

yash431garg opened the pull request:
- Description
Added new user tab with feature of giving user credit with their wallet address. 
- What are the steps to test that this code is working?
Try giving credit to a user id
- Screen shots or recordings for UI changes
<img width="1280" alt="Screenshot 2025-01-24 at 5 17 58 PM" src="https://github.com/user-attachments/assets/8909aa51-6473-4553-8e89-98e980644522" />

yash431garg commented:
@marvinmarnold I haven't added script right now. Could you confirm me that the schema change is good than I can utilize in the script as well?

samott commented on app/admin/users/page.tsx:
```diff
@@ -0,0 +1,94 @@
+  } = useForm({ resolver: zodResolver(userSchema) });
+
+  const onSubmit = handleSubmit(async (data) => {
+    window.confirm(`Give ${data.wallet} user ${data.credits} credits?`);
```
Does this handle the "no" case?

yash431garg replied:
Yea,

yash431garg replied:
There is some bug I will check again.

samott commented on scripts/distribute-credit.ts:
```diff
@@ -0,0 +1,66 @@
+                  userId: t.userId,
+                  asset: FungibleAsset.Credit,
+                  change: creditAmount,
+                  type: TransactionLogType.PremiumQuestionCharge,
```
Shouldn't this be `CreditByAdmin`?

yash431garg replied:
Yes, I was testing in the replica and we don't have the migration in that db branch. That's why I changed it this for testing and forgot to update. Thanks for pointing it. I have updated this.

samott reviewed (commented):
Looks good overall, works well.

samott commented:
Please rebase your branches to get the test fixes when you get a chance.

marvinmarnold reviewed (commented):
Everything working well. 
![Screenshot 2025-01-27 at 08 57 18](https://github.com/user-attachments/assets/824ed90a-9738-44dd-84a3-8c9e12e116cd)

Except for vertical scroll in admin panel on some pages

marvinmarnold commented:
@yash431garg some pages are still squished: https://chomp-8wz7qxd4w-gator-labs.vercel.app/admin/decks
![Screenshot 2025-01-27 at 10 34 18](https://github.com/user-attachments/assets/2b9079ad-6131-4d66-b1da-ed8727ff5c79)


And it looks like bottom padding might be gone in other spots.

![Screenshot 2025-01-27 at 10 34 50](https://github.com/user-attachments/assets/7f6a0411-6e0e-4d48-a863-fac428127d3a)

yash431garg commented:
> @yash431garg some pages are still squished: https://chomp-8wz7qxd4w-gator-labs.vercel.app/admin/decks ![Screenshot 2025-01-27 at 10 34 18](https://private-user-images.githubusercontent.com/1311030/407004311-2b9079ad-6131-4d66-b1da-ed8727ff5c79.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc5OTYxOTYsIm5iZiI6MTczNzk5NTg5NiwicGF0aCI6Ii8xMzExMDMwLzQwNzAwNDMxMS0yYjkwNzlhZC02MTMxLTRkNjYtYjFkYS1lZDg3MjdmZjVjNzkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjdUMTYzODE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTE5YTEzMmZmNTgyNjJkNzlhMWY2YzBkNjBjNWY5MDUxM2Y3OTlkYzVjMzUxMWRjOWNhYjkyYTcwM2FmYTYwMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.IK1JveGRNuUHc-cllA49uf_HFXvzJUwaDmOrham4FyI)
> 
> And it looks like bottom padding might be gone in other spots.
> 
> ![Screenshot 2025-01-27 at 10 34 50](https://private-user-images.githubusercontent.com/1311030/407004486-7f6a0411-6e0e-4d48-a863-fac428127d3a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc5OTYxOTYsIm5iZiI6MTczNzk5NTg5NiwicGF0aCI6Ii8xMzExMDMwLzQwNzAwNDQ4Ni03ZjZhMDQxMS02ZTBlLTRkNDgtYTg2My1mYWM0MjgxMjdkM2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjdUMTYzODE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzcwMjk5NDJkMjg3YzhkMTI3ZDRjMWVmNzc5MjJmYWMyOThkOGJhZGQzZWI1MjQ2ZWZkOWI4ZDcyYzYyZDRmMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.BT_zDjVorxtJUXItIuKgWQ-cVRDuO53sYAlr36V5px4)

Yes, It seems to be fine in local. But causing weird behavior in preview.

yash431garg commented:
@marvinmarnold I have removed overflow property for now. Should I pick this up as separate small fix later. It will cause this in mobile screen but solve the desktop layout
<img width="1280" alt="Screenshot 2025-01-27 at 10 23 36 PM" src="https://github.com/user-attachments/assets/84d2b9f9-1653-4cf9-aa25-cc3d167fbad5" />

marvinmarnold commented:
> @marvinmarnold I have removed overflow property for now. Should I pick this up as separate small fix later. It will cause this in mobile screen but solve the desktop layout issue. 

Yes, that's fine. You can also try removing 'Tags' tab since we aren't using it. I think that will just impact the likelihood of seeing the horizontal scroll error though. It won't be a permanent fix.

yash431garg commented:
> > @marvinmarnold I have removed overflow property for now. Should I pick this up as separate small fix later. It will cause this in mobile screen but solve the desktop layout issue.
> 
> Yes, that's fine. You can also try removing 'Tags' tab since we aren't using it. I think that will just impact the likelihood of seeing the horizontal scroll error though. It won't be a permanent fix.

I have fixed the issue as a part of this ticket.
